### ISSUE TITLE
Container crashes after one hour

### ISSUE THREAD

**Issue by ivan-perl** [progaudi/tarantool-docker#4]
Hello, 

We are working on a project where Tarantool used as a central storage. We are using progaudi images because of spacer which is missing in official images. Also our solution is based on .Net Core 2.1

Currently we are facing an issue that container stops after one hour with 100% reproducability. This happens even when our application is not running. Just starting container, waiting 3600 seconds and it disappears.

Docker inspect tells us that container ExitCode was 139. Alwasy, after each failure. Using our jenkins we made about 30 experiments on two different hosts. Accoring to this:
https://stackoverflow.com/questions/46724237/docker-exiting-with-status-code-139

Code 139 may mean Signal 11 or SIGSEGV

One of our fuesses was that this caused by creation of snapshot, because by default this happens every 3600 seconds. We updated configuration of our deployment like this:

`box.cfg{checkpoint_interval=180}`

and logs confirms that setting was applied and snapshots are created correctly:
`
2018-12-01 18:24:29.642 [1] main/108/checkpoint_daemon I> scheduled the next snapshot at Sat Dec  1 18:27:29 2018`
`2018-12-01 18:27:29.643 [1] main/108/checkpoint_daemon I> making snapshot...`
`2018-12-01 18:27:29.643 [1] main/108/checkpoint_daemon I> scheduled the next snapshot at Sat Dec  1 18:30:29 2018`
`2018-12-01 18:30:29.644 [1] main/108/checkpoint_daemon I> making snapshot...`
`2018-12-01 18:30:29.645 [1] main/108/checkpoint_daemon I> scheduled the next snapshot at Sat Dec  1 18:33:29 2018
`

Docker version is 17.09.1-ce, build f4ffd2511ce9
OS: OpenSuSE 42.3 LTS

uname details:
Linux linux-ubc0.suse 4.4.138-59-default #1 SMP Mon Jun 18 13:48:42 UTC 2018 (f0b8f6b) x86_64 x86_64 x86_64 GNU/Linux

Our docker-compose.yml is:
`version: '2'`

`services:`
`    tarantool:`
`        container_name: mytarantool`
`        image: progaudi/tarantool:2.1.1`
`        command: tarantool /usr/local/share/tarantool/app.init.lua`
`    ports:`
`        - 3301:3301`
`    volumes:`
`        - /extra/jenkins/workspace/insight_deploySurvilanceTest/infraRepo/Docker/Tarantool-basic:/usr/local/share/tarantool`
`        - /extra/jenkins/workspace/insight_deploySurvilanceTest/tarantoolData:/var/lib/tarantool`
`        - /extra/jenkins/workspace/insight_deploySurvilanceTest/srcRepo/Persistence/Tarantool/Records:/var/lib/tarantool-migrations`

Have you ever seen anything like this? What kind of logs/diagnostics can we also gather?

In docker logs for this container there is nothing relate to moment of failure, just info about startup and notes about snapshot creations

We tried several image versions starting 2.0

Thank you in advance,

Regards,
Ivan

**Comment by aensidhe** [comment]
1. There was a [fix of curl crash](https://github.com/tarantool/docker/commit/0448f8017f5f24ea9cb69dc726904f9b50487017) in upstream that I just pulled in. Can you try tag `2.1.1-48-g071918421`, please?
2. Also, you can try to increase log level when your application starts.
3. You can try official image with addition of spacer (it's easy to install it) to pinpoint the problem (tarantool itself/docker infrastructure or something specific to this image)

**Comment by ivan-perl** [comment]
Hi Anatoly, 

Thank you for quick reply. I tried to follow your first suggestion and I already see that container survived over 3600 seconds:
`--------- Flight report ---------`
 `- Flight time : 4050 seconds`
 `- Server log size : `
 `- Server status is : ALIVE (0)`
 `- Tarantool status is : ALIVE (0)`
`---------------------------------`
I will keep it running for a while and will make few more runs. If everything will be fine - I'll close this issue today or tomorrow.

Thank you so much for your help

Regards,
Ivan

**Comment by ivan-perl** [comment]
Hi again,

We made some more testing and looks like this issue has gone with this new image
`--------- Flight report ---------`
` - Flight time         : 41385 seconds`
` - Server log size     : 15174`
` - Server status is    : ALIVE (0)`
` - Tarantool status is : ALIVE (0)`
---------------------------------`

we can close this issue, thank you for your assistance. One quick question, when this fix will be delivered in a release version of the container?

Regards,
Ivan

**Comment by aensidhe** [comment]
What do you mean by `release version `?

**Comment by ivan-perl** [comment]
I mean something like `2.1.2`, instead of `2.1.1-48-g071918421`. This is some engineering build, right? Not sure it is  the right thing to tie our deployment configruation to some engineering build

**Comment by aensidhe** [comment]
It is CI build from commit [071918421 ](https://github.com/tarantool/tarantool/commit/071918421)in tarantool repo.
`2.1.1-48-g071918421` is output of [`git describe`](https://github.com/progaudi/tarantool-docker/blob/83922d1b9d6d73130c5a1825e00f06783f5f5e23/build-image.sh#L46) and means: 48th commit with hash `071918421` after tag [2.1.1](https://github.com/tarantool/tarantool/releases/tag/2.1.1).

2.1.2 will be posted when that release will be done by Tarantool team.

As stated in [info](https://github.com/progaudi/tarantool-docker) about this repo, when `2.1.1-48-g071918421` was built, `2.1.1` and `2.1` tags of docker image were also updated. 

If you want specific release build, send me a hash of commit and I'll build it.

**Comment by ivan-perl** [comment]
I just tried `2.1.1` and it crashed again after one hour. Are you sure that changes made in `2.1.1-48-g071918421` already available in 2.1.1 tag?

I have following in my docker file:
`image: progaudi/tarantool:2.1.1`
instead of working
`image: progaudi/tarantool:2.1.1-48-g071918421`
that worked fine

**Comment by aensidhe** [comment]
@ivan-perl it's already unavailable in that tag (4 hours ago was `2.1.1-57-gc6e5bf481` built. https://hub.docker.com/r/progaudi/tarantool/tags/). It's continuous integration build on every commit to tarantool repository. If you want to pin version, you should use full tag, not 2.1.1.

Also, if you had 2.1.1 image locally before testing `2.1.1-48-g071918421` and did not pull it again, you'll use old image from cache that does not have fix.

**Comment by ivan-perl** [comment]
Thank you @aensidhe , I tried one more time with following config:
`image: progaudi/tarantool:2.1.1`
with cleanup for gradle caches and container survived over night:
`--------- Flight report ---------`
` - Flight time : 43830 seconds`
` - Server log size : 15174`
` - Server status is : ALIVE (0)`
` - Tarantool status is : ALIVE (0)`
`---------------------------------`

And it is still alive.

Looks like we are good now. Closing this issue.

Thank you so much for your assistance!
